### DISCUSSION TITLE
eglot, pyright and docker not working well

### DISCUSSION THREAD

**Discussion by nettoyoussef** [joaotavora/eglot#828] [Q&A]
Hi.

I am able to set up pyright inside docker, but only with the npm version (the `pip` version is not installing the `pyright-langserver` command for some reason). 
So, inside the container I have to install both npm (``apt install npm``), and then, pyright (``npm -g install pyright``).
Using this method, once inside the container, `eglot` is able to open the language server using both `tramp` and `docker-tramp`.
For example: `/docker:root@container:/path/on/the/container`.

However no help documentation is available with this server, while, with the same process, they are with the `pylsp` server.
Basically,  I get "Import could not be resolved" for most libraries. 
This is probably a configuration problem regarding paths.
Since I am using an Ubuntu image, my user's python is on `/usr/local/bin/python3`, and I think pyright is using the 
image system-wide python installation on `/usr/bin/python3` (which leads to not resolving the imports).
I am using the official docker image `python:3.9.7`.

Any suggestions on what should I try to make this work?
